app/src/main/kotlin/PushYourLuck/DiceClass.kt

```diff
@@ -0,0 +1,58 @@
+package PushYourLuck
+
+import StatelessSolver
+import kotlin.random.Random
+
+class DiceClass(val nDice: Int,
+                val nSides: Int,
+                var markedSides: MutableList<MutableList<Boolean>> = Array(nDice){ i -> Array(nSides){i -> false}.toMutableList() }.toMutableList(),
```

Comment:
Huh? This is a poor use of Kotlin primitives and there must be a more concise way of writing this.